be/src/olap/rowset/beta_rowset_reader.cpp

```diff
@@ -88,7 +88,7 @@ Status BetaRowsetReader::init(RowsetReaderContext* read_context) {
     // load segments
     RETURN_NOT_OK(SegmentLoader::instance()->load_segments(
             _rowset, &_segment_cache_handle,
-            read_context->reader_type == ReaderType::READER_QUERY));
```

Comment:
for compaction, reader_type is not READER_QUERY, it does not use cache?

Reply:
Yes, here just make sure it follow the option and to prevent misuse. Since the reader type default value is *READER_QUERY*.